### PR TITLE
[WIP]Make wiki/note page headings more minimal, based on the new style guide

### LINKED ISSUE #5956: Make wiki/note page headings more minimal, based on the new style guide
@gautamig54 @CleverFool77 I was working in the [style guide draft](https://publiclab.org/notes/warren/05-07-2019/introducing-a-draft-style-guide-for-public-lab), and have been trying to see how we can keep a minimum of extra stuff in the heading, as it's currently so overloaded: 

Screen Shot 2019-06-27 at 7 34 10 PM

So I tried putting everything in a `...` menu just under the title. But then i thought, it's bit too hidden maybe? So I tried a row of buttons in the sidebar:

image

Any thoughts? Ideas? Preferences?

### PR DESCRIPTION
**Pull request by Ugtan**
Fixes #5956

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Ugtan** [comment]
@publiclab/reviewers Please have a look. I will also attach the screenshots below. Thanks

*Notes*
Screenshot from 2019-11-09 20-15-23

*Wiki*
Screenshot from 2019-11-09 20-14-08

**Comment by jywarren** [comment]
Hi this is awesome! We're you able to move the extra items into one of the menus instead? Thanks!

**Comment by Ugtan** [comment]
Hey @jywarren, did you mean move the extra stuff that is removed like views, comments, etc of the wiki and notes to the sidebar button on the top right corner?

**Comment by jywarren** [comment]
That's right - the round button with the `...` icon. That way the functions are still available, they're just a bit more hidden! 

Also note how we put the "edit" button in a circular button on pages like this: https://publiclab.org/notes/warren/11-30-2017/build-a-papercraft-spectrometer-for-your-phone-version-2-0

It'd be great to do the same with wiki pages like this! Do you think you could try implementing that?

**Comment by jywarren** [comment]
These could fit into this dropdown here: 

image

I think the code for this section is here: https://github.com/publiclab/plots2/blob/master/app/views/like/_like.html.erb

**Comment by Ugtan** [comment]
> It'd be great to do the same with wiki pages like this! Do you think you could try implementing that?

I can definately do that.

Okay. Thank you so much for the details. I will implement it soon.

**Comment by jywarren** [comment]
Thanks, that's awesome! Just ping me if you need any help!

**Comment by jywarren** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

**Comment by Ugtan** [comment]
> Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

Hey, I'm not stuck. I'm just having my exams so I'm a bit busy. But, i will try to do it by this weekend hopefully. Thanks

**Comment by jywarren** [comment]
Hi, looks good! Can you upload a screenshot or gif of the menu open? Thanks!

**Comment by Ugtan** [comment]
> Hi, looks good! Can you upload a screenshot or gif of the menu open? Thanks!
Merge state

Here you go.
image
image

**Comment by jywarren** [comment]
This looks great. Now it looks like a few tests that had been expecting to find "views" listed in the old location are now failing. Could you try adjusting the tests to look in the new location? The problem may be that the popups are constructed in JavaScript so they may not be actually visible to the functional tests (i.e. they don't appear in plain HTML). So we could remove them, or test for something else, or we could move them to system tests, which do run JavaScript.

```

ERROR["test_tag_blog", #<Minitest::Reporters::Suite:0x0000555f5c43b410 @name="TagControllerTest">, 5.811496078000005]
 test_tag_blog#TagControllerTest (5.81s)
ActionView::Template::Error:         ActionView::Template::Error: undefined method `views' for nil:NilClass
            app/views/like/_like.html.erb:92:in `_app_views_like__like_html_erb___306867212391491333_46934030695660'
            app/views/tag/blog.html.erb:65:in `block in _app_views_tag_blog_html_erb__3819743556353517424_46934031720500'
            app/views/tag/blog.html.erb:46:in `each_with_index'
            app/views/tag/blog.html.erb:46:in `_app_views_tag_blog_html_erb__3819743556353517424_46934031720500'
            test/functional/tag_controller_test.rb:353:in `block in <class:TagControllerTest>'
```

I guess the simplest seems to me to be just changing them to test for some of the new header text you've reduced things down to, rather than the text that is now hidden in the popover. What do you think?

**Comment by jywarren** [comment]
Oh you know, actually this kind of looks like it's not failing in the view, but actually somehow we're passing a nil into the tests and they're trying to get `nil.views` instead of a node. Can you take a look at the error carefully and see if any change you made would have caused this? All three test failures reference the same kind of error.

**Comment by Ugtan** [comment]
> Oh you know, actually this kind of looks like it's not failing in the view, but actually somehow we're passing a nil into the tests and they're trying to get nil.views instead of a node. Can you take a look at the error carefully and see if any change you made would have caused this? All three test failures reference the same kind of error.
Okay. I will try to figure it out.

File: app/views/like/_like.html.erb [hunk 1]
```diff
@@ -81,7 +86,26 @@
     
         <br />
         <hr style='clear:both;' />
-    
+        <div>
+            <span>
+                <% if node.type == 'note' %>
+                    <%= number_with_delimiter(@node.views) %> <%= t('notes.show.views') %>
```
Comment (jywarren) [inline app/views/like/_like.html.erb]: Aha - here, i think we just need `node.views` and not `@node.views`! I'm going to try making this change to see if it works!!

**Comment by jywarren** [comment]
Hooray!!! This looks great. The test failure was pretty simple - just removing `@` from some lines. Thanks a ton!!!

**Comment by Ugtan** [comment]
> Hooray!!! This looks great. The test failure was pretty simple - just removing @ from some lines. Thanks a ton!!!

That was easy! Thank you so much for helping me out. :)

**Comment by jywarren** [comment]
No problem! Thank you!!

**Pull request merged**